### PR TITLE
@ #40 | fix rsync for linux host

### PR DESCRIPTION
**Pull request by hieptranquoc**
connect #40 
please check this PR! Thank you!

### CODE REVIEW

**Comment by hieptranquoc** [comment]
I need to change this PR, I will assign it when I finish

**Comment by hieptranquoc** [comment]
@hoatle I've updated! Please check this PR! Thank you!

File: lib/teracy-dev-essential/config/rsync_recovery.rb [hunk 1]
```diff
@@ -7,6 +7,22 @@ module Config
...
+
+        if Vagrant::Util::Platform.linux?
+          if TeracyDev::Util.require_version_valid?(vagrant_version, ">= 2.2.3")
+            return unless rsync?(settings)
+
+            @cmd = 'rsync-auto'
+
+            @last_node = settings['nodes'].last['name']
+
+            return
```
Comment (hoatle) [inline lib/teracy-dev-essential/config/rsync_recovery.rb]: and you're removing all the existing logic, this kind of code is messy and smelly to me, it's very hard to check when the logic changes very often.

For example, `$ vagrant gatling-rsync-auto` will not auto switch to `$ vagrant up` anymore.

So please move the changes of this issue as a condition check only on the trigger stuff (configure_trigger_after function) instead.

**Comment by hieptranquoc** [comment]
@hoatle I've updated! Please check this PR again! Thank you...

File: lib/teracy-dev-essential/config/rsync_recovery.rb [hunk 2]
```diff
@@ -110,6 +121,14 @@ def valid_command(plugins)
             exec "vagrant up"
           end
         end
+
+        rsync_cmd = ARGV.any? { |s| s.include?('rsync-auto') }
+
+        if rsync_cmd
+          @logger.warn("'vagrant rsync-auto' does support rsync recovery, will use 'vagrant up' instead")
```
Comment (hoatle) [inline lib/teracy-dev-essential/config/rsync_recovery.rb]: display this warning only if it's not Linux and vagrant version < 2.2.3

File: lib/teracy-dev-essential/config/rsync_recovery.rb [hunk 3]
```diff
@@ -110,6 +121,14 @@ def valid_command(plugins)
             exec "vagrant up"
           end
         end
+
+        rsync_cmd = ARGV.any? { |s| s.include?('rsync-auto') }
+
+        if rsync_cmd and Vagrant::Util::Platform.linux? and TeracyDev::Util.require_version_valid?(Vagrant::VERSION, "< 2.2.3")
```
Comment (hoatle) [inline lib/teracy-dev-essential/config/rsync_recovery.rb]: if rsync_cmd and TeracyDev::Util.require_version_valid?(Vagrant::VERSION, "< 2.2.3")

**Comment by hieptranquoc** [comment]
@hoatle I applied your comment! Please check again! Thank you so much!

**Pull request merged**